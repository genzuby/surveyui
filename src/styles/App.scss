@import url("https://fonts.googleapis.com/css?family=Work+Sans&display=swap");

:root {
  --active--yellow: #f88f07;
  --light--yellow: #f7a02e;
  --available--bg: #44d350;
  --available--font: #fff;
  --available--border: #349c3d;
  --disable--bg: #e9e9e9;
  --disable--font: #9e9e9e;
  --disable--border: #c7c7c7;
  --text--main--color: #262626;
  --bg--dark--color: #241d4b;
  --bg--gray--color: #888888;
  --bg--gray--light: rgb(245, 245, 245);
  --bg--gray: rgb(235, 235, 235);
}

body,
html {
  font-family: "Work Sans", sans-serif;
  margin: 0;
  padding: 0;
  overflow: hidden;
  min-width: 320px;
}

.App {
  height: 100vh;
  position: relative;

  .SurveyBody {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 94%;
    height: 96vh;
    border-radius: 1em;
    box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.2);
    display: flex;
    align-items: center;
    flex-direction: column;

    .Header {
      height: 35%;
      .Heading {
        padding: 2em;
        h2 {
          text-align: center;
        }
      }
      .Content {
        padding: 2em 1.5em;

        p {
          text-align: center;
          font-size: 1.2rem;
        }
      }
    }

    .TypeComp {
      height: 40%;
      width: 100%;
      display: flex;
      align-items: flex-start;
      justify-content: center;
      flex-direction: column;
    }

    .Buttons {
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: row;
      width: 95%;
      height: 25%;
    }
  }
}
