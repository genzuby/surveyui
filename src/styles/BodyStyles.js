/* eslint-disable */
import styled from "styled-components";

// SurveyBody BG MOBILE
export const BODY = styled.div`
  background-color: ${props =>
    props.first
      ? "rgba(0, 0, 0, 0.3)"
      : props.last
      ? "rgba(0, 0, 0, 0.55)"
      : ""};
  background: ${props =>
    props.first
      ? "url(/images/welcome.jpg)"
      : props.last
      ? "url(/images/thankyou.jpg)"
      : "var(--bg--gray--light)"};
  background-blend-mode: multiply;
  background-size: cover;
  background-repeat: no-repeat;
  color: ${props =>
    props.first ? "#fff" : props.last ? "#fff" : "var(--text--main--color)"};
  border: ${props => (props.first ? "0px" : props.last ? "0px" : "1px")} solid
    var(--bg--gray);
  p,
  h1 {
    text-shadow: ${props =>
      props.first
        ? "2px 2px 2px rgba(0,0,0,0.3)"
        : props.last
        ? "2px 2px 2px rgba(0,0,0,0.3)"
        : ""};
  }
`;

// SurveyBody BG WEB
export const BODYWEB = styled.div`
  background-color: ${props =>
    props.first
      ? "rgba(0, 0, 0, 0.3)"
      : props.last
      ? "rgba(0, 0, 0, 0.55)"
      : ""};
  background: ${props =>
    props.first
      ? "url(/images/welcome-web.jpg)"
      : props.last
      ? "url(/images/thankyou-web.jpg)"
      : "var(--bg--gray--light)"};
  background-blend-mode: multiply;
  background-size: cover;
  background-repeat: no-repeat;
  color: ${props =>
    props.first ? "#fff" : props.last ? "#fff" : "var(--text--main--color)"};
  border: ${props => (props.first ? "0px" : props.last ? "0px" : "1px")} solid
    var(--bg--gray);
  p,
  h1 {
    text-shadow: ${props =>
      props.first
        ? "2px 2px 2px rgba(0,0,0,0.3)"
        : props.last
        ? "2px 2px 2px rgba(0,0,0,0.3)"
        : ""};
  }
`;

// Previous / Nest Button
export const BUTTON = styled.button`
  width: 100%;
  background: ${props =>
    props.available ? "var(--available--bg)" : "var(--disable--bg)"};
  border: 1px solid
    ${props =>
      props.available ? "var(--available--border)" : "var(--disable--border)"};
  height: 2.6em;
  font-size: 1.1rem;
  color: ${props =>
    props.available ? "var(--available--font)" : "var(--disable--font)"};
  border-radius: 4px;
  margin: 0 0.2em;
  outline: none;
  /* cursor disable when it's not available */
  pointer-events: ${props => (!props.available ? "none" : "auto")};
  cursor: ${props => (!props.available ? "nnot-not-allowed" : "pointer")};
  text-align: center;
`;

// Input styles
export const INPUT = styled.input`
  width: 80%;
  height: 2.3em;
  font-size: 1.1rem;
  padding: 0 0.5em;
  border: 1px solid var(--bg--gray--color);
  position: absolute;
  left: 50%;
  transform: translate(-50%);
`;

export const INPUROTH = styled.div`
  padding: ${props => (props.type === "radio" ? ".8em 15%" : "1em")};
  label {
    font-size: 1rem;
  }
`;

export const INPUROTHWEB = styled.div`
  padding: ${props =>
    props.type === "radio"
      ? ".8em calc(25% + 5vw)"
      : "1em 0 1em calc(10% + 10vw)"};
  label {
    font-size: 1rem;
  }
`;
