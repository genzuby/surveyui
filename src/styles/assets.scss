@import "App.scss";

// Checkbox and radio input styling
input[type="checkbox"] {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  border: 0;
}
input[type="checkbox"] + label {
  display: inline-block;
  position: relative;
  cursor: pointer;
}
input[type="checkbox"] + label:before {
  content: " ";
  display: inline-block;
  width: 21px;
  height: 21px;
  line-height: 21px;
  margin: -2px 8px 0 0;
  text-align: center;
  vertical-align: middle;
  border: 1px solid var(--light--yellow);
  border-radius: 2px;
}

input[type="checkbox"] + label {
  padding-left: 30px;
}
input[type="checkbox"] + label:before {
  position: absolute;
  left: 0;
  top: 0;
  margin-top: 0;
  opacity: 0.6;
  box-shadow: none;
  border-color: var(--light--yellow);
  transition: all 0.12s, border-color 0.08s;
}

input[type="checkbox"]:checked + label:before {
  position: absolute;
  content: "";
  width: 10px;
  top: -5px;
  left: 5px;
  border-radius: 2px;
  opacity: 1;
  background: transparent;
  border-color: transparent var(--light--yellow) var(--light--yellow)
    transparent;
  border-top-color: transparent;
  border-left-color: transparent;
  transform: rotate(10deg);
  transition: transform 0.1s;
}

input[type="checkbox"]:checked + label:before {
  content: "\2714";
  top: 0;
  left: 0;
  width: 21px;
  line-height: 21px;
  color: var(--active--yellow);
  text-align: center;
  border: 0;
}

input[type="radio"] {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  border: 0;
}
input[type="radio"] + label {
  display: inline-block;
  position: relative;
  padding-left: 30px;
  cursor: pointer;
}
input[type="radio"] + label:before {
  content: "";
  position: absolute;
  left: 0;
  top: -4px;
  width: 21px;
  height: 21px;
  text-align: center;
  background: #fff;
  border: 1px solid var(--light--yellow);
  border-radius: 100%;
}

input[type="radio"]:checked + label:before {
  background: #fff;
  border-color: var(--light--yellow);
}
input[type="radio"]:checked + label:after {
  content: "";
  position: absolute;
  top: 1px;
  left: 5px;
  width: 13px;
  height: 13px;
  background: var(--active--yellow);
  border-radius: 100%;
}
