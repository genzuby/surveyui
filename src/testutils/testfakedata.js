export const surveylist = [
  {
    questionID: "page-welcome",
    type: "Page",
    heading: "Welcome to the survey",
    body: "You can start the survey by clicking the next button",
    next: "question-email"
  },
  {
    questionID: "page-uneligible",
    type: "Page",
    heading: "Not eligible",
    body: "We're sorry, but you are not eligible to use our producs/services"
  },
  {
    questionID: "page-finish",
    type: "Page",
    heading: "Thank you",
    body: "You have completed the survey"
  },
  {
    questionID: "question-email",
    type: "Text",
    question: "What is your email address?",
    next: "question-age"
  },
  {
    questionID: "question-age",
    type: "Radio",
    question: "Are you 18 or older?",
    choices: [
      {
        label: "Yes",
        value: "yes"
      },
      {
        label: "No",
        value: "no",
        next: "page-uneligible"
      }
    ],
    next: "question-problems"
  },
  {
    questionID: "question-problems",
    type: "Checkbox",
    question: "Do you suffer from any of the following problems?",
    choices: [
      {
        label: "I have dental implants",
        value: "dental_implants"
      },
      {
        label: "I have missing teeth",
        value: "missing_teeth"
      },
      {
        label: "I have unfilled cavities",
        value: "unfilled_cavities"
      },
      {
        label: "I have swollen or inflamed gums",
        value: "swollen_or_inflamed_gums"
      },
      {
        label: "I have retainers",
        value: "retainers"
      },
      {
        label: "None of the above",
        value: "no"
      }
    ],
    next: "page-finish"
  }
];

export const initData = [
  {
    surveyID: "345f52c7-78f6-4cc4-9f8a-0490744f7da3",
    surveyItems: [
      {
        questionID: "page-welcome",
        type: "Page",
        heading: "Welcome to the survey",
        body: "You can start the survey by clicking the next button",
        next: "question-email"
      },
      {
        questionID: "page-uneligible",
        type: "Page",
        heading: "Not eligible",
        body:
          "We're sorry, but you are not eligible to use our producs/services"
      },
      {
        questionID: "page-finish",
        type: "Page",
        heading: "Thank you",
        body: "You have completed the survey"
      },
      {
        questionID: "question-email",
        type: "Text",
        question: "What is your email address?",
        next: "question-age"
      },
      {
        questionID: "question-age",
        type: "Radio",
        question: "Are you 18 or older?",
        choices: [
          {
            label: "Yes",
            value: "yes"
          },
          {
            label: "No",
            value: "no",
            next: "page-uneligible"
          }
        ],
        next: "question-problems"
      },
      {
        questionID: "question-problems",
        type: "Checkbox",
        question: "Do you suffer from any of the following problems?",
        choices: [
          {
            label: "I have dental implants",
            value: "dental_implants"
          },
          {
            label: "I have missing teeth",
            value: "missing_teeth"
          },
          {
            label: "I have unfilled cavities",
            value: "unfilled_cavities"
          },
          {
            label: "I have swollen or inflamed gums",
            value: "swollen_or_inflamed_gums"
          },
          {
            label: "I have retainers",
            value: "retainers"
          },
          {
            label: "None of the above",
            value: "no"
          }
        ],
        next: "page-finish"
      }
    ]
  }
];

export const testAnswer = {
  userId: "test3333@gmail.com",
  userInfo: {
    under18: false,
    finished: true
  },
  answeredlist: {
    surveyID: "345f52c7-78f6-4cc4-9f8a-0490744f7da3",
    surveyItems: [
      {
        questionID: "question-problems",
        type: "Checkbox",
        choices: ["dental_implants", "unfilled_cavities"]
      }
    ]
  }
};
