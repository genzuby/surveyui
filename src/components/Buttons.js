import React from "react";
// for Adapted web design
import { isMobile, isBrowser } from "react-device-detect";
import { BUTTON } from "../styles/BodyStyles";

const styles = {
  width: "calc(50% - 20vw)"
};

const Buttons = ({ first, last, onHandle }) => {
  //  when first is true then previous button disable
  //  when last is true then next button disable
  const { onPrevButtonHandle, onNextButtonHandle } = onHandle;

  const renderWeb = () => {
    if (isMobile) {
      return (
        <>
          <BUTTON
            available={!first}
            onClick={onPrevButtonHandle}
            data-test="prevBtnComponent"
          >
            ◁ Previous
          </BUTTON>
          <BUTTON
            available={!last}
            onClick={onNextButtonHandle}
            data-test="nextBtnComponent"
          >
            Next ▷
          </BUTTON>
        </>
      );
    } else if (isBrowser) {
      return (
        <>
          <BUTTON
            available={!first}
            onClick={onPrevButtonHandle}
            style={styles}
            data-test="prevBtnComponent"
          >
            ◁ Previous
          </BUTTON>
          <BUTTON
            available={!last}
            onClick={onNextButtonHandle}
            style={styles}
            data-test="nextBtnComponent"
          >
            Next ▷
          </BUTTON>
        </>
      );
    }
  };
  return <div className="Buttons">{renderWeb()}</div>;
};

export default Buttons;
