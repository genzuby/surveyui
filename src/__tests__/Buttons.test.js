import React from "react";
import { mount } from "enzyme";
import "jest-styled-components";
import "../testutils/setup";
import { findByTestAttr } from "../testutils";
import Buttons from "../components/Buttons";
import { BUTTON } from "../styles/BodyStyles";

describe("Buttons Component", () => {
  describe("Render Buttons", () => {
    let wrapper;
    let prevFunc, nextFnc;
    let first, last;

    beforeEach(() => {
      prevFunc = jest.fn();
      nextFnc = jest.fn();
      const props = {
        first,
        last,
        onHandle: {
          onPrevButtonHandle: prevFunc,
          onNextButtonHandle: nextFnc
        }
      };
      wrapper = mount(<Buttons {...props} />);
    });

    describe("Snapshot test", () => {
      it("renders correctly", () => {
        expect(wrapper).toMatchSnapshot();
      });

      it("should render two <button>", () => {
        expect(wrapper.find("button")).toHaveLength(2);
      });

      it("Previous button at the frist page should not be available", () => {
        wrapper.setProps({ first: true, last: false });
        const button = findByTestAttr(wrapper.find(BUTTON), "prevBtnComponent");

        expect(button).toHaveStyleRule("background", "var(--disable--bg)");
      });

      it("Next button at the frist page should be available ", () => {
        wrapper.setProps({ first: true, last: false });
        const button = findByTestAttr(wrapper.find(BUTTON), "nextBtnComponent");
        expect(button).toHaveStyleRule("background", "var(--available--bg)");
      });

      it("Previous button at the middle page should not be available", () => {
        wrapper.setProps({ first: false, last: false });
        expect(wrapper.find(BUTTON)).toHaveStyleRule(
          "background",
          "var(--available--bg)"
        );
      });

      it("Next button at the middle page should be available ", () => {
        wrapper.setProps({ first: false, last: false });
        expect(wrapper.find(BUTTON)).toHaveStyleRule(
          "background",
          "var(--available--bg)"
        );
      });

      it("Previous button at the last page should not be available", () => {
        wrapper.setProps({ first: true, last: true });
        expect(wrapper.find(BUTTON)).toHaveStyleRule(
          "background",
          "var(--disable--bg)"
        );
      });

      it("Next button at the last page should not be available", () => {
        wrapper.setProps({ first: true, last: true });
        expect(wrapper.find(BUTTON)).toHaveStyleRule(
          "background",
          "var(--disable--bg)"
        );
      });
    });
  });
});
