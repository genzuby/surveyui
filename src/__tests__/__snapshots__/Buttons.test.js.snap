// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Component Render Buttons Snapshot test renders correctly 1`] = `ReactWrapper {}`;
